api/restHandler/CoreAppRestHandler.go

```diff
@@ -1587,6 +1587,16 @@ func (handler CoreAppRestHandlerImpl) createWorkflows(ctx context.Context, appId
 }
 
 func (handler CoreAppRestHandlerImpl) createWorkflowInDb(workflowName string, appId int, userId int32) (int, error) {
+	//checking if workflow name is already exist or not
+	workflow, err := handler.appWorkflowRepository.FindByNameAndAppId(workflowName, appId)
+	if err != nil && !errors.Is(err, pg.ErrNoRows) && !errors.Is(err, pg.ErrMultiRows) {
+		handler.logger.Errorw("error in finding workflow by app id and name", "name", workflowName, "appId", appId)
+		return 0, err
+	}
+	// if workflow name already exists then we will assign a new name to the workflow
+	if workflow.Id != 0 {
+		workflowName = fmt.Sprintf("%s-clone-%s", workflowName, util.Generate(4))
```

Comment:
make a func for this, maybe something GenerateNewWorkflowName, also use this everywhere